### PR TITLE
chore: bump to nightly-2023-08-23

### PR DESCRIPTION
**Pull request by semorrison**
`nightly-2023-08-19` is now the tentative release candidate for the first official release of Lean 4, so it would be nice to try it out explicitly everywhere!

### CODE REVIEW

**Comment by mhuisi** [comment]
I'll merge this when #15 is merged :)

**Comment by mhuisi** [comment]
Is this still the current release candidate?

File: lean-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-leanprover/lean4:nightly-2023-08-15
+leanprover/lean4:nightly-2023-08-19
```
Comment (semorrison) [inline lean-toolchain:1]:
```suggestion
leanprover/lean4:nightly-2023-08-23
```

**Comment by semorrison** [comment]
It is now! :-)

**Pull request merged**